### PR TITLE
feat: Support HTTP Basic Authentication in REST Swagger connector generator

### LINKED ISSUE #436: Implement support for HTTP Basic Authentication
For the API connector we need HTTP Basic Authorization support. When generating a custom connector we need to add properties for username/password that can be set when a connection is generated for the connector.

### PR DESCRIPTION
**Pull request by zregvart**
Adds support for HTTP Basic Authentication (not digest!) in REST Swagger connector generator.

Fixes #436

### CODE REVIEW

File: app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java [hunk 1]
```diff
@@ -76,13 +89,24 @@ public void setAuthenticationType(final AuthenticationType authenticationType) {
...
 
+    public void setUsername(final String username) {
+        this.username = username;
+    }
+
     /* default */ void addAuthenticationHeadersTo(final Map<String, Object> headers) {
         if (authenticationType == AuthenticationType.oauth2) {
             headers.put("Authorization", "Bearer " + accessToken);
+        } else if (authenticationType == AuthenticationType.basic) {
+            final String usernameAndPassword = username + ":" + password;
```
Comment (jimmidyson) [inline app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java]: Do we care if username has `:` in it?

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
     private final String specification;
 
     public BaseSwaggerGeneratorExampleTest(final String connectorQualifier, final String name) throws IOException {
-        specification = resource("/swagger/" + name + ".swagger.json");
+        specification = resource("/swagger/" + name + ".swagger.json", "/swagger/" + name + ".swagger.yaml");
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java]: Supporting YAML definition is separate to this PR, but I'm OK to include it here or have it in separate PR - me likey granular PRs :)

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/TestHelper.java [hunk 3]
```diff
@@ -56,4 +56,12 @@ private TestHelper() {
         return resource;
     }
 
+    /* default */ static String resource(final String path, final String alternative) throws IOException {
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/TestHelper.java]: I like this but how about expanding it even more, varargs loop through until find first resource? Slightly more flexible and might help us in future.

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java (hunk 2 above)
Comment (jimmidyson) [reply app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java]: At least in a separate commit anyway please.

File: app/rest/connector-generator/src/test/resources/swagger/petstore.param_connector.json [hunk 4]
```diff
@@ -172,11 +172,11 @@
   },
   "actions": [
     {
-      "id": "io.syndesis:rest-swagger-connector:1.2-SNAPSHOT:_id_:updatePet",
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:updatePet",
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/test/resources/swagger/petstore.param_connector.json]: Would be good to do this is separate commit please - needed change but easier to track in separate commit as it's unrelated to core change of this PR.

File: app/rest/connector-generator/src/test/resources/swagger/petstore.unified_connector.json [hunk 5]
```diff
@@ -185,11 +185,11 @@
   },
   "actions": [
     {
-      "id": "io.syndesis:rest-swagger-connector:1.2-SNAPSHOT:_id_:updatePet",
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:updatePet",
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/test/resources/swagger/petstore.unified_connector.json]: Same comment - separate commit please, well same commit as other similar changes.

File: app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java (hunk 1 above)
Comment (zregvart) [reply app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java]: Probably, I don't think connector is proper place to make such decisions, it should take the property on face value. Not sure if there is a way to enforce property syntax for configured properties. Created #841

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java (hunk 2 above)
Comment (zregvart) [reply app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java]: We already parse Swagger files using the Swagger API that supports both JSON and YAML, so this doesn't provide additional functionality. It was just somewhat easier to craft a YAML Swagger file for this test.

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/TestHelper.java (hunk 3 above)
Comment (zregvart) [reply app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/TestHelper.java]: Let's deal deal with that in the future, right now we need only one alternative (we check for JSON or YAML file present).

File: app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java (hunk 1 above)
Comment (jimmidyson) [reply app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java]: This code path could be reached from multiple places so being defensive here makes sense to me as this is creating the basic auth header.

However... from https://tools.ietf.org/html/rfc2617#section-2 colon isn't valid in username so i think we can assume here that it is safe.

File: app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java (hunk 2 above)
Comment (jimmidyson) [reply app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java]: Ah OK.
